### PR TITLE
Propagated parameters to beta

### PR DESCRIPTION
**Pull request by chitrangpatel**
# Changes
[Propagated Parameters](#4955) was deployed as an alpha feature in release [v0.36.0](https://github.com/tektoncd/pipeline/releases/tag/v0.36.0). This PR promotes it to beta by removing the constraints applied under the `alpha` feature. This PR addresses issue https://github.com/tektoncd/pipeline/issues/4955.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Promote propagated parameters to beta.
```

### CODE REVIEW

**Comment by abayer** [comment]
/hold

Holding this pending beta/stable/etc discussions in other places, such as #5530 and #5515. Probably will get covered at the API WG on Monday.

File: pkg/apis/pipeline/v1/task_validation_test.go
Comment (jerop) [inline pkg/apis/pipeline/v1/task_validation_test.go:145]: we need to check if beta api fields is enabled in v1

**Review by jerop — COMMENTED** [review]
thank you @chitrangpatel!

we need to guard the propagation behind beta feature flag in v1 - https://github.com/tektoncd/pipeline/pull/5325

**Comment by jerop** [comment]
/hold cancel

we agreed to support alpha/beta/stable to unblock progress - https://github.com/tektoncd/pipeline/pull/5325

File: pkg/apis/pipeline/v1/task_validation_test.go
Comment (chitrangpatel) [reply pkg/apis/pipeline/v1/task_validation_test.go:145]: sounds good!

File: pkg/apis/pipeline/v1beta1/taskrun_validation_test.go
Comment (jerop) [inline pkg/apis/pipeline/v1beta1/taskrun_validation_test.go:66]: we do not need this check in v1beta1 - see https://github.com/tektoncd/pipeline/pull/5628#discussion_r993724518

File: pkg/reconciler/taskrun/validate_resources.go
Comment (jerop) [inline pkg/reconciler/taskrun/validate_resources.go:140]: this is not relevant to the promotion, right? i recommend that this goes in a separate commit & PR

**Review by jerop — CHANGES_REQUESTED** [review]
believe the "extra params" change should be in a separate commit & PR

File: pkg/reconciler/taskrun/validate_resources.go
Comment (chitrangpatel) [reply pkg/reconciler/taskrun/validate_resources.go:140]: Leaving this in makes the tests fail. It seems like the default action was to check if there are extra parameters and throw an error. When the alpha feature flag was enabled, this would not be the case. Now that propagated params is being promoted to beta, we cannot leave this filter in just for "alpha", we need to do the same for "beta" as well. That's why I thought it belonged in the same PR.

Comment (chitrangpatel) [reply pkg/reconciler/taskrun/validate_resources.go:140]: I'm happy to open a separate PR just for this and fix it there. However, it will have to be merged in for propagated params to beta to work.

File: docs/pipelineruns.md
Comment (JeromeJu) [inline docs/pipelineruns.md:329]: Sorry for running late on this review.
Wondering for this do we also need to add it to the [list of beta features](https://github.com/tektoncd/pipeline/blob/main/docs/install.md#beta-features)?

Comment (jerop) [reply docs/pipelineruns.md:329]: didn't see a list there and thought we'd create the list when we make the v1 release

**Pull request merged**
